src/libpmem2/x86_64/memset/memset_nt_sse2.h

```diff
@@ -160,4 +148,6 @@ EXPORTED_SYMBOL(char *dest, int c, size_t len)
 	memset_small_sse2(dest, xmm, len);
 end:
 	maybe_barrier();
+
+	VALGRIND_DO_FLUSH(orig_dest, orig_len);
```

Comment:
Does this have to be after the flush? We could move `VALGRIND_DO_FLUSH` before any dest/len modifications which would make this simpler.

Reply:
well, this function modifies the data... calling VALGRIND_DO_FLUSH before would be a bit pointless ;)

Reply:
right xD